astrosat_users/serializers/serializers_auth.py

```diff
@@ -49,10 +50,19 @@ def validate(self, attrs):
         but that uses this serializer for the user and so validation will be checked when processing the view.)
         """
         self.instance = super().validate(attrs)
+        user = self.instance["user"]
 
         adapter = get_adapter(self.context.get("request"))
         try:
-            adapter.check_user(self.instance["user"])
+            # the user might change accepted_terms...
+            accepted_terms = self.initial_data.get("accepted_terms", None)
+            if accepted_terms is not None:
+                accepted_terms_field = self.fields["accepted_terms"]
+                accepted_terms_value = accepted_terms_field.to_internal_value(accepted_terms)
```

Comment:
```suggestion
                accepted_terms_value = self.fields["accepted_terms"].to_internal_value(accepted_terms)
```
The `accepted_terms_field isn't used anywhere else, so do we need it assigned to a variable. If you feel what is there is easier to read though, I'm not against it.

Reply:
You're right it's not needed.  I did it to make it easier to read.